Sources/ApolloTestSupport/TestMock.swift

```diff
@@ -75,6 +75,26 @@ public class Mock<O: MockObject>: AnyMock, Hashable {
     _data[field.key.description] = value?._unsafelyConvertToMockValue()
   }
 
+  public subscript<T: AnyScalarType & Hashable>(
+    dynamicMember keyPath: KeyPath<O.MockFields, Field<Array<T>>>
+  ) -> [T]? {
+    get {
+      let field = O._mockFields[keyPath: keyPath]
+      return _data[field.key.description] as? [T]
+    }
+    set {
+      _setList(newValue, for: keyPath)
+    }
+  }
+
+  public func _setList<T: AnyScalarType & Hashable>(
```

Comment:
I think we should rename this ti `setScalarList`. The point of defining these methods in the first place was to avoid the issues we were having with trying to make sure the compiler always used the right overload for the functions based on the types. Just to maintain that pattern, they should all have unique names.